src/main/java/javax/measure/spi/ServiceProvider.java

```diff
@@ -383,7 +383,7 @@ private static ServiceProvider setDefault(ServiceProvider provider) {
     public static final ServiceProvider setCurrent(ServiceProvider provider) {
         Objects.requireNonNull(provider);
         ServiceProvider old = current.getAndSet(provider);
-        if (old != provider) {
+        if (old != null && !old.equals(provider)) {
```

Comment:
Could (optionally - not a big deal) be replaced by `provider.equals(old)` since `provider` is guaranteed non-null by above `requireNonNull(provider)` call, and `x.equals(null)` should return `false` as per `Object.equals(Object)` specification.

However another question is whether we really want to use `Object.equals(Object)` in this particular case. If we use `ServiceProvider.equals(Object)` method, maybe Unit-API specification would need to define what is the contract for this method, i.e. how we define equality between two `ServiceProvider` instances? Currently, the specification said nothing about that.

Reply:
For some reason, my mind fooled me with the impression that you explicitly asked for the null check the traditional way. Revisiting the comments proved me wrong and I adjusted it to the way you said, which is also the way I prefer, so, we have an agreement.

Now, regarding the second part of your comment, I think @keilw has already provided his view above. That said, I leave it up to you, Werner, whether this conversation has to be resolved now (which implies merging this PR as well) or not (which implies that we need some further discussion/clarification on that)